test/unit/proc_macro/proc_macro_test.bzl

```diff
@@ -0,0 +1,100 @@
+"""Unittest to verify proc-macro targets"""
+
+load("@bazel_skylib//lib:unittest.bzl", "analysistest")
+load("//rust:defs.bzl", "rust_proc_macro", "rust_test")
+load("//test/unit:common.bzl", "assert_action_mnemonic", "assert_list_contains_adjacent_elements", "assert_list_contains_adjacent_elements_not")
+
+def _proc_macro_test_targets(edition):
+    """Define a set of `rust_proc_macro` targets for testing
+
+    Args:
+        edition (str): The rust edition to use for the new targets
+    """
+    rust_proc_macro(
+        name = "proc_macro_lib_{}".format(edition),
+        srcs = [
+            "proc_macro_{}.rs".format(edition),
+        ],
+        edition = edition,
+    )
+
+    rust_test(
+        name = "proc_macro_lib_{}_unittest".format(edition),
```

Comment:
Nit: rename unittest -> `wrapper_rule_for_macro_{}`?

Reply:
I think I preferred it before but not strongly enough to not make the change 😅